### DISCUSSION TITLE
Execute pre- and/or post-import SQL

### DISCUSSION THREAD

**Discussion by iboates** [osm2pgsql-dev/osm2pgsql#2283] [Ideas]
I have a workflow in which I have to execute SQL after my import to do things that aren't possible or feasible in the flex style file. I was just thinking it would be nice to have a means of executing arbitrary custom SQL before or after import. Maybe by command line parameters?

**Comment by joto** [comment]
You can always just use `psql` to do that.

The other option is using `osm2pgsql-gen` which has functionality to simply run SQL commands. This allows also to run SQL commands for each tile (of zoom zoom level that you specify) which can really be useful when running a query on the whole planet is not feasible or when doing updates and only some areas with changes needs recomputing. `osm2pgsql-gen` was created for generalization of map data, but it can really be used for all sorts of post-processing. At some point we want to integrate this functionality into osm2pgsql itself, so you don't need to run two programs, but the details are not clear yet.
